SQF-VM/SQF_commands.c

```diff
@@ -1499,89 +1386,113 @@ void CMD_SELECT(void* input, CPCMD self)
 	right = pop_stack(vm, vm->work);
 	left_val = get_value(vm, vm->stack, left);
 	right_val = get_value(vm, vm->stack, right);
-	if (left_val == 0 || right_val == 0)
-	{
+	if (left_val == 0 || right_val == 0) {
 		inst_destroy(left);
 		inst_destroy(right);
 		return;
 	}
 	//ToDo: implement more select variants https://community.bistudio.com/wiki/select
 
-
-	if (left_val->type == ARRAY_TYPE())
-	{
+	if (left_val->type == ARRAY_TYPE()) {
 		arr = left_val->val.ptr;
-		if (right_val->type == SCALAR_TYPE())
-		{
+		if (right_val->type == SCALAR_TYPE()) {
 			index = roundf(right_val->val.f);
-		}
-		else if (right_val->type == BOOL_TYPE())
-		{
+		} else if (right_val->type == BOOL_TYPE()) {
 			index = right_val->val.i > 0 ? 1 : 0;
-		}
-		else
-		{
+		} else {
 			vm->error(vm, ERR_RIGHT_TYPE ERR_SCALAR ERR_OR ERR_BOOL, vm->stack);
-			push_stack(vm, vm->stack, inst_value(value(NOTHING_TYPE(), base_int(0))));
+			push_stack(vm, vm->stack,
+					inst_value(value(NOTHING_TYPE(), base_int(0))));
 			inst_destroy(left);
 			inst_destroy(right);
 			return;
 		}
-	}
-	else
-	{
+	} else {
 		vm->error(vm, ERR_LEFT_TYPE ERR_ARRAY, vm->stack);
-		push_stack(vm, vm->stack, inst_value(value(NOTHING_TYPE(), base_int(0))));
+		push_stack(vm, vm->stack,
+				inst_value(value(NOTHING_TYPE(), base_int(0))));
 		inst_destroy(left);
 		inst_destroy(right);
 		return;
 	}
 
-	if (index < 0 || index > arr->top)
-	{
+	if (index < 0 || index > arr->top) {
 		vm->error(vm, ERR_SPECIAL_SELECT_1, vm->stack);
-		push_stack(vm, vm->stack, inst_value(value(NOTHING_TYPE(), base_int(0))));
+		push_stack(vm, vm->stack,
+				inst_value(value(NOTHING_TYPE(), base_int(0))));
 		inst_destroy(left);
 		inst_destroy(right);
 		return;
-	}
-	else if (index == arr->top)
-	{
-		push_stack(vm, vm->stack, inst_value(value(NOTHING_TYPE(), base_int(0))));
-	}
-	else
-	{
+	} else if (index == arr->top) {
+		push_stack(vm, vm->stack,
+				inst_value(value(NOTHING_TYPE(), base_int(0))));
+	} else {
 		tmp = arr->data[index];
 		push_stack(vm, vm->stack, inst_value(value(tmp->type, tmp->val)));
 	}
 	inst_destroy(left);
 	inst_destroy(right);
 }
 
+void CMD_SELECTRANDOM(void* input, CPCMD self) {
+	PVM vm = input;
+	PINST right;
+	PVALUE right_val;
+	PARRAY arr;
+	PVALUE tmp;
+	right = pop_stack(vm, vm->work);
+	right_val = get_value(vm, vm->stack, right);
+	if (right_val == 0) {
+		inst_destroy(right);
+		return;
+	}
+
+	if (!(right_val->type == ARRAY_TYPE())) {
+		vm->error(vm, ERR_RIGHT_TYPE ERR_ARRAY, vm->stack);
+		push_stack(vm, vm->stack,
+				inst_value(value(NOTHING_TYPE(), base_int(0))));
+		inst_destroy(right);
+		return;
+	}
+
+	arr = right_val->val.ptr;
+
+	if (arr->top == 0) {
+		push_stack(vm, vm->stack,
+				inst_value(value(NOTHING_TYPE(), base_int(0))));
+		inst_destroy(right);
+		return;
+	}
+
+	srand(time(NULL));
+	int index = rand() % arr->top;
```

Comment:
Please declare all used variables in the functions begining before first non-declaring line

Reply:
Doesn't this actually initializes the variables with an unneeded value that gets overwritten as soon as the variable is being used?